src/aws-cpp-sdk-identity-management/source/auth/STSProfileCredentialsProvider.cpp

```diff
@@ -306,6 +312,11 @@ void STSProfileCredentialsProvider::Reload()
     }
 
     m_credentials = profileIt->second.GetCredentials();
+    
+    if (!m_credentials.IsEmpty()) {
```

Comment:
instead of updating `m_credentials` after the fact, why not update the credentials before we set the member variable when we knows its resolved from the source profile? i.e.

```cpp
const auto arn = sourceProfiles.back()->second.GetRoleArn();
const auto& assumedCreds = GetCredentialsFromSTS(stsCreds, arn);

if (!assumedCreds.IsEmpty()) {
  assumedCreds.AddUserAgentFeature(Aws::Client::UserAgentFeature::CREDENTIALS_PROFILE_SOURCE_PROFILE);
}

sourceProfiles.back()->second.SetCredentials(assumedCreds);
```